### PR TITLE
Refactor the plugin into more methods to allow future extension

### PR DESCRIPTION
**Pull request by Cadair**
Hopefully, this increases readability, I am planning on adding hash comparisons on top of this.

### CODE REVIEW

**Comment by astrofrog** [comment]
@Cadair - can you rebase for the CI?

File: pytest_mpl/plugin.py [hunk 1]
```diff
@@ -202,17 +202,138 @@ def __init__(self,
         if self.results_dir and not os.path.exists(self.results_dir):
             os.mkdir(self.results_dir)
 
+    def get_compare(self, item):
```
Comment (astrofrog) [inline pytest_mpl/plugin.py:205]: Can you add a docstring for this?

**Review by astrofrog — CHANGES_REQUESTED** [review]
Very nice refactoring - can you check whether the astropy image test suite passes with this?

**Comment by Cadair** [comment]
@astrofrog I have opened a PR on astropy to see if the figure tests pass with this branch.

**Comment by Cadair** [comment]
Astropy build passed :tada:

**Review by astrofrog — APPROVED** [review]
Looks good to me assuming the ci passes!

**Pull request merged**